bindgen/extension.bzl

```
@@ -0,0 +1,24 @@
+"""Module extension for accessing a bindgen toolchain"""
+
+load(":repositories.bzl", "rust_bindgen_dependencies")
+load(":transitive_repositories.bzl", "rust_bindgen_transitive_dependencies")
+
+def _bindgen_impl(_):
+    rust_bindgen_transitive_dependencies()
+    rust_bindgen_dependencies()
+    bindgen_toolchains(
+        name = "bindgen_toolchains",
+        build = "//bindgen/toolchain:BUILD.bazel",
+    )
+
+bindgen = module_extension(implementation = _bindgen_impl)
+
+def _bindgen_toolchains_impl(ctx):
+    ctx.file("BUILD.bazel", ctx.read(ctx.attr.build))
+
+bindgen_toolchains = repository_rule(
+    implementation = _bindgen_toolchains_impl,
+    attrs = {
+        "build": attr.label(mandatory = True),
```

I'm not sure if this is the case for bzlmod but isn't the precedent something like
```suggestion
        "build_file_content": attr.string(mandatory = True),
        "build_file": attr.label(mandatory = True),
```

---

Replaced with `local_repository`

---

Reverted that, added the `build_file_content` parameter.